presto-main/src/main/java/com/facebook/presto/operator/scalar/CombineHashFunction.java

```diff
@@ -24,6 +24,6 @@ private CombineHashFunction() {}
     @SqlType(StandardTypes.BIGINT)
     public static long getHash(@SqlType(StandardTypes.BIGINT) long previousHashValue, @SqlType(StandardTypes.BIGINT) long value)
     {
-        return 31 * previousHashValue + value;
+        return (int) (31 * previousHashValue + value);
```

Comment:
Add a comment explaining why a function that returns a `long` truncates the value to an `int`.